crates/sui-indexer/src/store/module_resolver_v2.rs

```diff
@@ -58,3 +65,58 @@ impl ModuleResolver for IndexerStoreModuleResolver {
             .cloned())
     }
 }
+
+/// InterimModuleResolver consists of a backup ModuleResolver
+/// (e.g. IndexerStoreModuleResolver) and an in-mem package cache.
+pub struct InterimModuleResolver<GM>
+where
+    GM: GetModule<Item = Arc<CompiledModule>, Error = anyhow::Error>,
+{
+    backup: GM,
+    package_cache: Arc<Mutex<InMemPackageCache>>,
+    metrics: IndexerMetrics,
+}
+
+impl<GM> InterimModuleResolver<GM>
+where
+    GM: GetModule<Item = Arc<CompiledModule>, Error = anyhow::Error>,
+{
+    // TODO remove this after integration is done
+    #[allow(dead_code)]
+    pub fn new(
+        backup: GM,
+        package_cache: Arc<Mutex<InMemPackageCache>>,
+        new_packages: &[IndexedPackage],
+        checkpoint_seq: CheckpointSequenceNumber,
+        metrics: IndexerMetrics,
+    ) -> Self {
+        package_cache
+            .lock()
+            .unwrap()
+            .insert_packages(new_packages, checkpoint_seq);
+        Self {
+            backup,
+            package_cache,
+            metrics,
+        }
+    }
+}
+
+impl<GM> GetModule for InterimModuleResolver<GM>
+where
+    GM: GetModule<Item = Arc<CompiledModule>, Error = anyhow::Error>,
+{
```

Comment:
nit: do you need the constraints on `GM` anywhere except on the `GetModule` impl?

```suggestion
pub struct InterimModuleResolver<GM> {
    backup: GM,
    package_cache: Arc<Mutex<InMemPackageCache>>,
    metrics: IndexerMetrics,
}

impl<GM> InterimModuleResolver<GM> {
    // TODO remove this after integration is done
    #[allow(dead_code)]
    pub fn new(
        backup: GM,
        package_cache: Arc<Mutex<InMemPackageCache>>,
        new_packages: &[IndexedPackage],
        checkpoint_seq: CheckpointSequenceNumber,
        metrics: IndexerMetrics,
    ) -> Self {
        package_cache
            .lock()
            .unwrap()
            .insert_packages(new_packages, checkpoint_seq);
        Self {
            backup,
            package_cache,
            metrics,
        }
    }
}

impl<GM> GetModule for InterimModuleResolver<GM>
where
    GM: GetModule<Item = Arc<CompiledModule>, Error = anyhow::Error>,
{
```

Reply:
Oh nice catch, yes we should only need to constraint on the GetModule impl

Reply:
thanks for pointing it out!